### ISSUE TITLE
Problem to convert gif transparent to png

### ISSUE THREAD

**Issue by anderolimar** [lovell/sharp#525]
Hi !
I had a problem when I tried to convert gif with transparent background to png.
The png background generated is white instead transparent.

```
sharp('input.gif')
       .resize(240, 240)
       .background({ r: 255, g: 255, b: 255, a: 0 })
       .crop(sharp.gravity.center)
       .toFormat(sharp.format.png)
       .toFile('output.png', function(err) {
      }); 
```

This behavior didn't happen in version 0.13.1

Bellow is the image url that I did test.

[http://25.media.tumblr.com/c99a579db3ae0fc164bf4cca148885d3/tumblr_mjgv8kEuMg1s87n79o1_400.gif](url)

![Image tested](http://25.media.tumblr.com/c99a579db3ae0fc164bf4cca148885d3/tumblr_mjgv8kEuMg1s87n79o1_400.gif)

Image output.

![Image output](https://raw.githubusercontent.com/anderolimar/travis.test/dev/output.png)

Thanks.

**Comment by lovell** [comment]
Hello, if you're using the default pre-compiled binaries provided by sharp then the change in behaviour is probably due to the [v0.15.0](http://sharp.dimens.io/en/stable/changelog/#v0150-21st-may-2016) switch from using *magick to giflib for GIF processing.

Based on the metadata of this image, it looks like its supposed to "dispose" onto a white background. I'm not sure what this means when there's also an alpha transparency channel.

```sh
$ identify -verbose tumblr_mjgv8kEuMg1s87n79o1_400.gif | grep Background
  Background Color: #FFFFFFFF
  Dispose: Background
  Background Color: white
  Dispose: Background
  Background Color: white
  Dispose: Background
  Background Color: white
  Dispose: Background
```

My best guess would be that one of either *magick or giflib are wrong, but I don't know enough about the GIF format to say which.

Are you able to install libvips with *magick support (then remove `node_modules/sharp` and reinstall) to see if this "fixes" it?

**Comment by anderolimar** [comment]
Ok, I will try this. Thank you for you awnser.

**Comment by lovell** [comment]
@anderolimar Were you able to make any progress with this?

**Comment by diegocsandrim** [comment]
Hi @lovell, I tried with @anderolimar to get it work like you said, but could not do it in Ubuntu.
Something I tried was using the CLI of libvips to convert the image, in a Windows machine, and that was the result:
Source Image (Gif with tranparence)
![image](https://cloud.githubusercontent.com/assets/4148815/17782284/2406693a-6549-11e6-9b86-77c554476fea.gif)

Using the command:
`./vips.exe copy image.gif copy-image.png`
The gif color that was set to transparenty was merged into the image with
![copy-image](https://cloud.githubusercontent.com/assets/4148815/17782332/5d16e88a-6549-11e6-81ae-58f351b34b3d.png)


Using the command
`./vips.exe magickload image.gif magick-image.png`
The transparence was retained
![magick-image](https://cloud.githubusercontent.com/assets/4148815/17782336/601dd52a-6549-11e6-99d2-21fe71ff32e5.png)

What do think, would it be an libvips error?

**Comment by lovell** [comment]
Thanks for the detailed response Diego. Assuming *magick has the correct behaviour, it looks like either giflib or libvips use/configuration of giflib that's causing the problem.

I'm not going to be able to help debug this for at least a couple of weeks so it's probably best to ask a question in the libvips repo to see if John or anyone else can help there.

**Comment by diegocsandrim** [comment]
Ok, I'll do it, thanks for your help!

**Comment by diegocsandrim** [comment]
Here is the issue in libvips: https://github.com/jcupitt/libvips/issues/502

**Comment by jcupitt** [comment]
Thanks to the nice test case, I found a bug in the gif loader. It should be fixed in 8.3 and master.

**Comment by lovell** [comment]
v0.17.0 now available (with a libvips dependency of v8.4.2+).
